presto-main/src/main/java/com/facebook/presto/server/testing/TestingPrestoServer.java

```diff
@@ -90,6 +97,42 @@
     private final ServiceSelectorManager serviceSelectorManager;
     private final Announcer announcer;
     private final QueryManager queryManager;
+    private final TaskManager taskManager;
+    private final GracefulShutdownHandler gracefulShutdownHandler;
+    private final ShutdownAction shutdownAction;
+    private final boolean coordinator;
+
+    public static class TestShutdownAction
+            implements ShutdownAction
+    {
+        private CountDownLatch shutdownCalled = new CountDownLatch(1);
+
+        @GuardedBy("this")
+        private boolean isWorkerShutdown = false;
+
+        @Override
+        public void onShutdown()
+        {
+            setWorkerShutdown(true);
+            shutdownCalled.countDown();
+        }
+
+        public void waitForShutdownComplete(long millis)
+                throws InterruptedException
+        {
+            shutdownCalled.await(millis, MILLISECONDS);
+        }
+
+        public synchronized boolean isWorkerShutdown()
+        {
+            return isWorkerShutdown;
+        }
+
+        public synchronized void setWorkerShutdown(boolean isWorkerShutdown)
```

Comment:
Since this isn't used outside this class, I would inline it and make `onShutdown` synchronized instead